backend/src/main/kotlin/app/ehrenamtskarte/backend/auth/service/Authorizer.kt

```diff
@@ -39,4 +39,27 @@ object Authorizer {
         return (user?.role == Role.REGION_ADMIN.db_value && user.regionId == region.id) ||
             mayViewUsersInProject(user, region.projectId.value)
     }
+
+    fun mayCreateAdministrator(
+        actingAdmin: AdministratorEntity,
+        newAdminProjectId: Int,
+        newAdminRole: Role,
+        newAdminRegion: RegionEntity?
+    ): Boolean {
+        if (actingAdmin.projectId.value != newAdminProjectId) {
+            return false
+        } else if (newAdminRole == Role.NO_RIGHTS) {
+            return false
+        }
+
+        if (actingAdmin.role == Role.PROJECT_ADMIN.db_value) {
+            return true
+        } else if (
+            actingAdmin.role == Role.REGION_ADMIN.db_value &&
+            newAdminRegion != null && actingAdmin.regionId == newAdminRegion.id
```

Comment:
We might want to add a check that only allows to set specific roles (regionAdmin and regionManager)

Reply:
Yes, I think currently a REGION_ADMIN can create a PROJECT_ADMIN?

I feel like these checks are becoming more and more complicated. A permission-based autorisation might fit better.

That is also what minecraft servers are using :D

Not sure if it is worth to refactor though. And I'm also not sure if we can express these complicated autorisation statements with simple permissions.

Reply:
Technically, he can't, because we don't allow the combination PROJECT_ADMIN + REGION_ID!=NULL
I will add a check nevertheless.

I think using permissions it wouldn't get easier to formulate these conditions.
I feel like the current ones are not too complex (probably because I've written them), but still they're all just a few lines (that should be tested :) )

Reply:
Haha yeah maybe tests would make the difference in terms of confidence.
If you trust all lines in Authorizer, then we don't need them right now though.